src/dart_array.rs

```diff
@@ -14,6 +14,22 @@ pub struct DartArray {
     inner: Box<[*mut DartCObject]>,
 }
 
+impl<T: IntoDart, const N: usize> From<[T; N]> for DartArray {
```

Comment:
Can we extract common code from `From<[T; N]> for DartArray` and `From<Vec<T>> for DartArray`? For example, by creating a `From<Iterable<T>> for DartArray`.

Currently there is duplicated code so it is not a very good practice imho